src/assets/scripts/client/changelog/ChangelogController.js

```diff
@@ -0,0 +1,200 @@
+import $ from 'jquery';
+import { SELECTORS } from '../constants/selectors';
+import { STORAGE_KEY } from '../constants/storageKeys';
+
+/**
+ * The controller class for the in-game changelog.
+ *
+ * @class ChangelogController
+ */
+export default class ChangelogController {
+    /**
+     * @constructor
+     * @param {ContentQueue} contentQueue
+     */
+    constructor(contentQueue) {
+        /**
+         * The content queue, used to load in the changelog data
+         *
+         * @property contentQueue
+         * @type {ContentQueue}
+         */
+        this.contentQueue = contentQueue;
+
+        /**
+         * The DOM of the changelog container, containing version, data, and dismiss.
+         *
+         * @property $changelogContainer
+         * @type {JQuery|HTMLElement}
+         */
+        this.$changelogContainer = $(SELECTORS.DOM_SELECTORS.CHANGELOG_CONTAINER);
+
+        /**
+         * The DOM element that displays the version above the changelog
+         *
+         * @property $changelogVersion
+         * @type {JQuery|HTMLElement}
+         */
+        this.$changelogVersion = $(SELECTORS.DOM_SELECTORS.CHANGELOG_VERSION);
+
+        /**
+         * Changelog data element - where the text goes
+         *
+         * @property $changelogData
+         * @type {JQuery|HTMLElement}
+         */
+        this.$changelogData = $(SELECTORS.DOM_SELECTORS.CHANGELOG_CONTENT);
+
+        /**
+         * The link at the bottom of the container to
+         * dismiss the changelog.
+         *
+         * @property $changelogDismiss
+         * @type {JQuery|HTMLElement}
+         */
+        this.$changelogDismiss = $(SELECTORS.DOM_SELECTORS.CHANGELOG_DISMISS);
```

Comment:
I am not a big fan of the dismiss button. None of our dialogs has a close / dismiss button and I would prefer to keep it consistent. Maybe we should add a "x" button to the top right corner of the dialogs (in another pull request).

Reply:
I agree it doesn't match the convention and would make sense to remove. I also agree it makes sense to at some point show this at initial load (put last loaded version in localstorage, and open changelog view when loaded version is newer than the one in localstorage).

So let's do this:
- create an issue for showing the changelog view when loading a version newer than the version noted in localstorage. Here we can determine how best to dismiss the window.
- Change the "dismiss" in this PR to "View Full Changelog", and link it to the changelog document, since people may miss more than 1 version, and wonder what changed since they last played

Reply:
@erikquinn The logic's already there for displaying the changelog if the version is new, should this be removed?

Reply:
Follow-up, should the link to the full changelog be pointed at the `master` or `develop` branch?

Reply:
neither.  I'd point it to the releases page: (https://github.com/openscope/openscope/releases)

Reply:
sounds good

Reply:
If it shows itself when a new version is loaded, then great!